app/src/main/java/com/delirium/playlistmaker/search/ui/activity/SearchFragment.kt

```diff
@@ -75,19 +87,19 @@ class SearchActivity : AppCompatActivity(), ClickListener {
             }
         }
 
-        viewModel.getHistoryLiveData().observe(this) { history ->
+        viewModel.getHistoryLiveData().observe(viewLifecycleOwner) { history ->
             renderHistory(history.asList())
         }
 
-        binding.recyclerSongs.layoutManager = LinearLayoutManager(this)
+        binding.recyclerSongs.layoutManager = LinearLayoutManager(requireContext())
         binding.recyclerSongs.adapter = adapter
 
         binding.clearSearch.setOnClickListener { it ->
             binding.editSearch.text.clear()
             it.visibility = View.INVISIBLE
 
-            this.currentFocus?.let {
-                val keyboard = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+            requireActivity().currentFocus?.let {
```

Comment:
🍏  Чтобы решить проблему с необходимостью двойного нажатия на поле ввода для показа клавиатуры, используй данные о фокусе самой вьюхи, а не всей активити, ведь у тебя на экране могут быть несколько фокусируемых объектов:

```suggestion
            if (binding.editSearch.isFocused) {
```